rp2040-hal/src/timer.rs

```diff
@@ -381,3 +381,49 @@ impl_alarm!(Alarm3 {
     int_name: "TIMER_IRQ_3",
     armed_bit_mask: 0b1000
 });
+
+/// Support for RTIC monotonic trait.
+#[cfg(feature = "rtic-monotonic")]
+pub mod monotonic {
+    use core::panic;
+
+    use super::Alarm;
+    use super::Instant;
+    use super::Timer;
+
+    /// RTIC Monotonic Implementation
+    pub struct Monotonic<A>(pub Timer, A);
+    impl<A: Alarm> Monotonic<A> {
+        /// Creates a new monotonic.
+        pub const fn new(timer: Timer, alarm: A) -> Self {
+            Self(timer, alarm)
+        }
+    }
+    impl<A: Alarm> rtic_monotonic::Monotonic for Monotonic<A> {
+        type Instant = Instant;
+        type Duration = fugit::MicrosDurationU64;
+
+        const DISABLE_INTERRUPT_ON_EMPTY_QUEUE: bool = false;
+
+        fn now(&mut self) -> Instant {
+            self.0.get_counter()
+        }
+
+        fn set_compare(&mut self, instant: Instant) {
+            if let Err(super::ScheduleAlarmError::AlarmTooLate) = self.1.schedule_at(instant) {
+                panic!("Instant too far in the future, it cannot be scheduled.");
```

Comment:
Is it really necessary to panic at this point?

I checked a few other implementations. Some just set the comparator value of the timer to the past without checking it at all, some do set the timer to the next tick. But I didn't find a single one which panics. 

I wonder what "Note: This method does not need to handle race conditions of the monotonic, the timer queue in RTIC checks this." in the trait docs really means, does it cover this situation?

Reply:
It turns out it is not. From what I understand, the interrupt may trigger any time before the instant and should trigger short after the target time (eg Systick-monotonic does not even do anything in set_compare because the interrupt gets triggered everytime anyways).

So I updated this method accordingly to have an intermediate trigger if the target instant is too far in the future.